pyteal/ast/subroutine.py

```diff
@@ -76,37 +78,176 @@ def __init__(
         self.declaration: Optional["SubroutineDeclaration"] = None
         self.__name = self.implementation.__name__ if nameStr is None else nameStr
 
+    @staticmethod
+    def check_if_ABI_class(arg_type: Type) -> bool:
+        return (
+            isclass(arg_type)
+            and arg_type is not abi.BaseType
+            and issubclass(arg_type, abi.BaseType)
+        )
+
+    @staticmethod
+    def check_if_ABI_annotation(abi_anno: Any) -> bool:
+        try:
+            abi.type_spec_from_annotation(abi_anno)
+            return True
```

Comment:
```suggestion
    def is_annotation(obj: Any) -> bool:
        try:
            abi.type_spec_from_annotation(obj)
            return True
```
similar to previous, but also recommend renaming param since we don't really no yet that it's an `abi_anno`